### DISCUSSION TITLE
Any way to access statically the list of icon names ?

### DISCUSSION THREAD

**Discussion by martin-cycle** [lucide-icons/lucide#1778] [Help] [answered]
### Package

- [X] lucide
- [ ] lucide-angular
- [ ] lucide-flutter
- [ ] lucide-preact
- [ ] lucide-react
- [ ] lucide-react-native
- [ ] lucide-solid
- [ ] lucide-svelte
- [ ] lucide-vue
- [ ] lucide-vue-next
- [ ] Figma plugin
- [ ] source/main
- [ ] other/not relevant

### Version

latest

### Browser

- [ ] Chrome/Chromium
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] iOS Safari
- [ ] Opera
- [ ] Other/not relevant

### Operating system

- [ ] Windows
- [ ] Linux
- [ ] macOS
- [ ] Other/not relevant

### Description

I would need to type the value of the icon used in the context of a graphql validator in a node express app.

### Steps to reproduce

/

### Checklist

- [X] I have searched if someone has submitted a similar issue before and there weren't any. (Please make sure to also search closed issues, as this issue might already have been resolved.)

**Comment by ericfennis** [comment, marked as answer]
You can use `lucide-static` package for this. In the root there are some files which includes list of all icon names.
See NPM file explorer: https://www.npmjs.com/package/lucide-static?activeTab=code, for example the `icon-nodes.json` file has object with names and icon nodes.

**Reply by martin-cycle** [reply]
Awesome, works like a charm 👌🏽

**Reply by hitecSmartHome** [reply]
but how it works?

**Reply by ericfennis** [reply]
@hitecSmartHome What is your use case?

**Reply by hitecSmartHome** [reply]
I want to create an icon picker component where i visually list all the available icons. For the love of god I can't figure out how to dynamically import all the icons

**Reply by hitecSmartHome** [reply]
@ericfennis Can you help me?

**Reply by ericfennis** [reply]
Did you already tried the dynamic imports?

**Reply by hitecSmartHome** [reply]
> Did you already tried the dynamic imports?

No, I dont know how

**Reply by spencermarx** [reply]
Hey Lucide team, hope y'all are well! 👋

I've been following this issue and wanted to chime in. First, I absolutely love Lucide - you've created an incredible icon library that's beautifully designed, consistently styled, and works seamlessly across frameworks. It's become my go-to for all projects!

I saw the suggestion to use `lucide-static` and the `icon-nodes.json` file. While that's a possible approach, it would still introduce performance tradeoffs similar to importing the entire library - additional package dependencies and potentially larger bundle sizes, which I'm trying to avoid in performance-sensitive applications.

What I think would be incredibly valuable is having first-class TypeScript types for icon names built directly into the main package. This could be easily implemented as part of your build process - a simple code generation step that creates a union type of all icon names right before publishing:

```typescript
export type LucideIconName = 'Activity' | 'AirVent' | 'Airplay' | /* all icons */;
```

This automated approach would ensure the type definitions stay in sync with the actual icons, while adding minimal overhead to the package. It would enable better TypeScript integration with features like autocompletion, type checking, and validation scenarios without requiring workarounds or additional dependencies.

In my specific case, I needed icon name validation in a schema but wanted to maintain optimal bundle size. I ended up creating a custom NX executor-based build step to generate these types, but I think having this as an official feature would benefit many developers in the ecosystem.

Really appreciate all your work on this library, and thanks for considering this enhancement!

**Comment by prpanto** [comment]
I really want the list of all the name of the icons.
@ericfennis what do you mean dynamic imports?

**Reply by karsa-mistmere** [reply]
Check out the accepted answer just above your question: https://github.com/lucide-icons/lucide/discussions/1778#discussioncomment-8023917

**Reply by karsa-mistmere** [reply]
> I really want the list of all the name of the icons.

Then use [`lucide-static`](https://lucide.dev/guide/packages/lucide-static) and import `tags.json` or `icon-nodes.json`

**Reply by karsa-mistmere** [reply]
> @ericfennis what do you mean dynamic imports?

Depends on what framework you're using, for React for example you can use this guide: https://lucide.dev/guide/packages/lucide-react#with-dynamic-imports

But be warned that we only recommend using dynamic imports if you actually need to dynamically use every single icon Lucide offers (e.g. if your app needs to implement an icon picker component or something along these lines).

**Reply by prpanto** [reply]
Is there any list of icon names?

**Reply by prpanto** [reply]
To explain... I use lucide icon and i cal them with vuetify icon with just pass the icon-#name_of_lucide and I want all the icons name to make a picker...

**Reply by karsa-mistmere** [reply]
> Is there any list of icon names?

Again: you can use `tags.json` or `icon-nodes.json` from `lucide-static`. If you want to implement a picker, you could also use the nodes from `icon-nodes.json` to manually render the SVGs from the picker, just as we do on lucide.dev or in our Figma plugin.
